sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -581,6 +588,15 @@ private Bound(String name, String filenamePrefix, String filenameSuffix,
        */
       public Bound<T> to(String filenamePrefix) {
         validateOutputComponent(filenamePrefix);
+        return new Bound<>(name, StaticValueProvider.of(filenamePrefix), filenameSuffix,
+            header, footer, coder, numShards, shardTemplate, validate,
+            writableByteChannelFactory);
+      }
+
+      /**
+       * Like {@code to()}, but with a {@link ValueProvider}.
```

Comment:
Generally, I think we'd prefer to copy the comment and change it rather than cross-reference. But it's not bad.